src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -658,13 +654,16 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
     XBU::verbose("Sub command: --base");
     std::string empty = "";
     if(update.compare("all") == 0)
-      auto_flash(deviceCollection, force);
-    else if(update.compare("flash") == 0)
-      throw xrt_core::error("Platform only update is not supported");
-    else if(update.compare("sc") == 0)
-      throw xrt_core::error("SC only update is not supported");
-    else 
+      auto_flash(deviceCollection, XBU::getForce());
+    else {
```

Comment:
why the split?

Reply:
A ladder of `else if` statements causes obfuscation and it if preferred to not have them.  So I refactored the code to see what the intent of the code base was.  In the process, I discovered that "flash" and "sc" are not needed and will be removing this code in the next code base.